internal/impl/io/input_socket_syslog.go

```diff
@@ -0,0 +1,252 @@
+		return
+	}
+	if p.socketType, err = conf.FieldString("type"); err != nil {
+		return
+	}
+
+	if p.address == "" {
+		p.log.Errorf("Field address must not be empty")
+		return
+	}
+	switch p.socketType {
+	case "unix":
+		p.streamingCn, err = p.createStreamingSocket()
+		if err != nil {
+			p.log.Errorf("Couldn't create socket: %v", err)
+		}
+	case "unixgram":
+		p.datagramCn, err = p.createDatagramSocket()
+		if err != nil {
+			p.log.Errorf("Couldn't create socket: %v", err)
+		}
+	}
+	return
+}
+
+func (p *syslogUnixReader) createStreamingSocket() (net.Listener, error) {
+	if _, err := os.Stat(p.address); err == nil {
+		if err := os.Remove(p.address); err != nil {
+			p.log.Errorf("Couldn't create socket: %v", err)
+		}
+	}
+	conn, err := net.Listen(p.socketType, p.address)
+	if err != nil {
+		return nil, err
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		os.Remove(p.address)
+		conn.Close()
+		os.Exit(1)
+	}()
+	return conn, nil
+}
+
+func (p *syslogUnixReader) createDatagramSocket() (*net.UnixConn, error) {
+	if _, err := os.Stat(p.address); err == nil {
+		if err := os.Remove(p.address); err != nil {
+			p.log.Errorf("Couldn't create socket: %v", err)
+		}
+	}
+
+	laddr, err := net.ResolveUnixAddr(p.socketType, p.address)
+	if err != nil {
+		return nil, err
+	}
+	unixConn, err := net.ListenUnixgram(p.socketType, laddr)
+	if err != nil {
+		return nil, err
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		os.Remove(p.address)
+		unixConn.Close()
+		os.Exit(1)
+	}()
+
+	return unixConn, nil
+}
+
+func (p *syslogUnixReader) recvDatagramMessage() (string, error) {
+	p.buf = make([]byte, p.maxBufferSize)
+	_, _, err := p.datagramCn.ReadFromUnix(p.buf)
+	if err != nil {
+		p.log.Errorf("Couldn't read Syslog message: &v", err)
+	}
+	msg, err := p.parseSyslogMessage(p.buf)
+	if err != nil {
+		return "", err
+	}
+	return msg, nil
+}
+
+func (p *syslogUnixReader) recvStreamingMessage() (string, error) {
+	p.buf = make([]byte, p.maxBufferSize)
+	conn, err := p.streamingCn.Accept()
+	if err != nil {
+		p.log.Errorf("Couldn't accept message on socket: ", err)
+	}
+
+	defer conn.Close()
+
+	_, err = conn.Read(p.buf)
+	if err != nil {
+		p.log.Errorf("Couldn't read Syslog message: &v", err)
+		return "", err
+	}
+
+	msg, err := p.parseSyslogMessage(p.buf)
+	if err != nil {
+		return "", err
+	}
+	return msg, nil
+}
+
+func (p *syslogUnixReader) parseSyslogMessage(buf []byte) (string, error) {
+	switch p.headerFormat {
+	case "rfc5424":
+		m := rfc5424.NewParser(buf)
+		if err := m.Parse(); err != nil {
+			return "", err
+		}
+		message := fmt.Sprintf("%v", m.Dump()["content"])
+		return message, nil
+	case "rfc3164":
+		m := rfc3164.NewParser(buf)
+		if err := m.Parse(); err != nil {
+			return "", err
+		}
+		message := fmt.Sprintf("%v", m.Dump()["content"])
```

Comment:
Would it make sense to pull out other bits of data and add them as metadata fields? In which case this method will need to return a message type instead.